miden/tests/integration/flow_control/mod.rs

```diff
@@ -277,3 +277,59 @@ fn simple_dyn_exec() {
         false,
     );
 }
+
+#[test]
+fn simple_dyncall() {
+    let program_source = "
+        proc.foo
+            # drop the top 4 values, since that will be the code hash when we call this dynamically
+            drop drop drop drop
+            add
+        end
+
+        begin
+            # call foo directly so it will get added to the CodeBlockTable
+            push.0 push.0 push.0 push.0
+            call.foo
+
+            # move the first result of foo out of the way
+            movdn.4
+
+            # use dyncall to call foo again via its hash, which is on the stack
+            dyncall
+        end";
```

Comment:
I would probably change the example slightly to test that `foo` is executed in a non-root context. Could be done in various ways, for example: write something to memory in the body of the program, then read something from the same memory address inside `foo` and make sure that what we read in `foo` is zeros.